Bug 1303412, Quota on Registries feature

ahardin-rh opened the pull request:
This is a follow-up to https://github.com/openshift/openshift-docs/pull/1599

 It addresses this bug:
https://bugzilla.redhat.com/show_bug.cgi?id=1303412

ahardin-rh commented:
@bfallonf @tpoitras can you please check behind me? Thanks!

tpoitras commented on dev_guide/builds.adoc:
```diff
@@ -1739,12 +1739,14 @@ manually)
+|A build fails with `requested access to the resource is denied`. What does it
+mean?
 |You have exceeded one of
 link:../dev_guide/quota.html#usage-limits[image quotas] set on your project.
```
s/one of/one of the

bfallonf commented on install_config/install/docker_registry.adoc:
```diff
@@ -171,17 +171,28 @@ endif::[]
-link:../../dev_guide/quota.html#create-a-quota[Create a quota object] on particular namespace with
-one or more of the following resources:
+link:../../dev_guide/quota.html#create-a-quota[Create a quota object] on
+particular namespace with one or more of the following resources:
```
"...on *a* particular namespace..."?

bfallonf commented on install_config/install/docker_registry.adoc:
```diff
@@ -171,17 +171,28 @@ endif::[]
+repository)
+
+|`openshift.io/projectimagessize`
+|To limit the total images size per project
```
@ahardin-rh I'm not too sure if I understand the above two statements. Could this be simpled down to:

"To limit an image's total size when tagged in an image stream (also known as a repository)"

"To limit an image's total size per projects"

Maybe?

tpoitras replied:
I thought the projectimagessize meant more like "To limit the total combined size of all images per project", but I could be wrong.

bfallonf commented:
@ahardin-rh Just a few suggestions/questions. Looks good otherwise!

tpoitras commented on install_config/install/docker_registry.adoc:
```diff
@@ -189,9 +200,8 @@ Once the quota is exceeded, an image push will fail with:
-You may then
-link:../../admin_guide/pruning_resources.html#pruning-images[prune obsoleted images]
-to retrieve some space back.
+Then, you may link:../../admin_guide/pruning_resources.html#pruning-images[prune
```
s/may/can

tpoitras commented on install_config/install/docker_registry.adoc:
```diff
@@ -189,9 +200,8 @@ Once the quota is exceeded, an image push will fail with:
-link:../../admin_guide/pruning_resources.html#pruning-images[prune obsoleted images]
-to retrieve some space back.
+Then, you may link:../../admin_guide/pruning_resources.html#pruning-images[prune
+obsoleted images] to retrieve some space back.
```
Unless I'm seeing this wrong in this GitHub UI, I would not put a line break between "prune" and "obsoleted images". This comes back to the discussion we had in this week's meeting about hard wrapping, where you don't want to hard wrap in the middle of a term, or link.

tpoitras commented:
A few nitpicks, otherwise good @ahardin-rh :+1: :octopus: :palm_tree:

ahardin-rh commented:
**revhistory note**

for link:../dev_guide/builds.html[Dev Guide --> Builds]

    Added a Troubleshooting Guidance table.

for link:../dev_guide/quota.html#usage-limits[Dev Guide --> Resource Quota]

    Added `openshift.io/imagesize`, `openshift.io/imagestreamsize`, and `openshift.io/projectimagessize` to the link:../dev_guide/quota.html#usage-limits[Usage Limits] table.

for link:../install_config/install/docker_registry.html[Deploying a Docker Registry]

    Added a link:../install_config/install/docker_registry.html#viewing-logs[Setting Storage Quota] section.
